runtime/src/accounts_index.rs

```diff
@@ -94,13 +94,22 @@ impl<'a, T: 'a + Clone> AccountsIndex<T> {
 
     // find the latest slot and T in a slice for a given ancestor
     // returns index into 'slice' if found, None if not.
-    fn latest_slot(&self, ancestors: Option<&Ancestors>, slice: SlotSlice<T>) -> Option<usize> {
+    fn latest_slot(
+        &self,
+        ancestors: Option<&Ancestors>,
+        slice: SlotSlice<T>,
+        max_root: Option<Slot>,
+    ) -> Option<usize> {
         let mut max = 0;
         let mut rv = None;
         for (i, (slot, _t)) in slice.iter().rev().enumerate() {
             if *slot >= max
```

Comment:
the interaction between `max_root` and `ancestors` are getting messier here. Is it desired/intuitive behavior to return  something newer than `max_root` if the ancestors isn't empty and contains the new slot?

Or, if `max_root` is given, it's guaranteed for `ancestors` to be empty? If so, maybe add a sanity assertion?

Reply:
Updated to be a little clearer, and added a comment, what do you think?